src/process.cc

```diff
@@ -119,6 +119,11 @@ Process::~Process() {
   while (has_messages()) {
     remove_first_message();
   }
+
+  Locker locker(OS::scheduler_mutex());
+  while (!root_certificates_.is_empty()) {
+    delete root_certificates_.remove_first();
+  }
```

Comment:
We also sometimes use:

```suggestion
  while (auto certificate = root_certificates_.remove_first()) {
    delete certificate;
  }
```